File: YuGiOh2/Cards/C52684508.cs
using System.Linq;
using YuGiOh2.Base;

namespace YuGiOh2.Cards
{
    /// <summary>
    /// 黑炎弹
    /// </summary>
    public class C52684508
    {
        public static int Type { get; } = (int)ChooseTargetType.None;

        public static bool CheckIfAvailable(Player player)
        {
            return player.Field.MonsterFields.Any(c => c != null && c.Password == "74677422" && !c.Status.FaceDown);
        }

        public static void ProcessEffect(Player player)
        {
            var monster = player.Field.MonsterFields
                .FirstOrDefault(c => c != null && c.Password == "74677422" && !c.Status.FaceDown);
            if (monster == null)
                return;
            player.Enemy.DecreaseHP(monster.ATK);
        }
    }
}

File: YuGiOh2/Scripts/C52684508.lua
Type = 0;
--黑炎弹
function checkIfAvailable(player)
    for index, monster in pairs(player.Field.MonsterFields) do
        if monster ~= nil and monster.Password == '74677422' and not monster.Status.FaceDown then
            return true;
        end
    end
    return false;
end

function processEffect(player)
    for index, monster in pairs(player.Field.MonsterFields) do
        if monster ~= nil and monster.Password == '74677422' and not monster.Status.FaceDown then
            player.Enemy:DecreaseHP(monster.ATK);
        end
    end
end